### ISSUE TITLE
How to train ADM-Baseline.pt on CiFar-10?

### ISSUE THREAD

**Issue by RachelTeamo** [forever208/DDPM-IP#21]
Thanks for the open source code, but I'm curious how to train to get ADM-Baseline on the CiFar-10 dataset. Is it just changing the perturbation to 0.0 in the settings below? I made a previous attempt when the FID was 7.14 after training 300K iters, so I would appreciate your advice. Thanks.
```
mpiexec -n 2  python scripts/image_train.py --input_pertub 0.15 \
--data_dir PATH_TO_DATASET \
--image_size 32 --use_fp16 True --num_channels 128 --num_head_channels 32 --num_res_blocks 3 \
--attention_resolutions 16,8 --resblock_updown True --use_new_attention_order True \
--learn_sigma True --dropout 0.3 --diffusion_steps 1000 --noise_schedule cosine --use_scale_shift_norm True \
--rescale_learned_sigmas True --schedule_sampler loss-second-moment --lr 1e-4 --batch_size 64
```
My train setting is 
```
CUDA_VISIBLE_DEVICES=0 torchrun --nproc_per_node=2 --master_port=23456 scripts/image_train.py \
--data_dir ./datasets/cifar_train --lr_anneal_steps 300000 \
--image_size 32 --use_fp16 False --num_channels 128 --num_head_channels 32 --num_res_blocks 3 --save_interval 50000 \
--attention_resolutions 16,8 --resblock_updown True --use_new_attention_order True \
--learn_sigma True --dropout 0.3 --diffusion_steps 1000 --noise_schedule cosine --use_scale_shift_norm True \
--rescale_learned_sigmas True  --schedule_sampler loss-second-moment  --lr 1e-4 --batch_size 64 --log_dir cifar
```
Since when I set use_fp16 True, The training loss will appear to be NaN.

**Comment by forever208** [comment]
@RachelTeamo hi, you are right, setting 'perturbation=0' is the baseline. Can you check that your batch_size is in total 128?

Except for using fp16, I think your parameter settings are correct. Can you try with fp16 because I don't have any nan when using fp16.

Additionally, use the whole training set (50k images) as the reference batch when you compute FID

**Comment by RachelTeamo** [comment]
ok, thanks. But I wonder about the --lr_anneal_steps 300000 of your settings?

**Comment by forever208** [comment]
@RachelTeamo I did not set --lr_anneal_steps as, just use its default value

**Comment by RachelTeamo** [comment]
The default value is a loop that does not exist, so are you using ema_0.9999_050000.pt as your baseline?

**Comment by forever208** [comment]
@RachelTeamo yes, `ema_0.9999_500000.pt` is the baseline model I trained and reported, it corresponds to the training of 500k steps using total batch size=128

**Comment by RachelTeamo** [comment]
OK, thanks again. I will have a try.
